subsystems/TypeHandlerLibrary/src/test/java/org/terasology/persistence/typeHandling/FutureTypeHandlerTest.java

```diff
@@ -77,4 +53,29 @@ public void testRecursiveType() {
 
         assertEquals(typeHandler, future.typeHandler);
     }
+
+    private static final class RecursiveType<T> {
+        final T data;
+        final List<RecursiveType<T>> children;
+
+        @SafeVarargs
+        private RecursiveType(T data, RecursiveType<T>... children) {
+            this.data = data;
+            this.children = Lists.newArrayList(children);
+        }
+    }
+
+    @edu.umd.cs.findbugs.annotations.SuppressFBWarnings
```

Comment:
why is this necessary?

Reply:
added a comment. message that one should consider making the class static. but the test has it non-static.

Reply:
> but the test has it non-static.

Sorry, you'll have to elaborate that a bit more, I don't follow.

Reply:
updated the comment @jdrueckert .